onnxruntime/core/providers/dml/DmlExecutionProvider/src/GraphDescBuilder.cpp

```diff
@@ -199,43 +222,52 @@
+            // map operatorGraphIntermediateEdges as mainGraphIntermediateEdge
+            for (auto& operatorGraphIntermediateEdge : graphNodeCreateInfo.intermediateEdges)
             {
-                if (graphNodeInfo.kernelOutputIndices[outputIndex] == std::numeric_limits<uint32_t>::max())
-                {
-                    continue;
-                }
+                DML_INTERMEDIATE_GRAPH_EDGE_DESC edge = {};
+                edge.FromNodeIndex = operatorGraphNodeIndexToMainGraphNodeIndexMap[operatorGraphIntermediateEdge.FromNodeIndex];
+                edge.FromNodeOutputIndex = operatorGraphIntermediateEdge.FromNodeOutputIndex;
+                edge.ToNodeIndex = operatorGraphNodeIndexToMainGraphNodeIndexMap[operatorGraphIntermediateEdge.ToNodeIndex];
+                edge.ToNodeInputIndex = operatorGraphIntermediateEdge.ToNodeInputIndex;
+                graphIntermediateEdges.push_back(edge);
+            }
 
-                uint32_t kernelOutputIndex = graphNodeInfo.kernelOutputIndices[outputIndex];
-                const onnxruntime::NodeArg* arg = node.OutputDefs()[kernelOutputIndex];
+            // populate nameToNodeAndIndexMap (which will be used by above loop) for operatorGraphOutputEdges
+            for (auto& operatorGraphOutputEdge : graphNodeCreateInfo.outputEdges)
+            {
+                const onnxruntime::NodeArg* arg = node.OutputDefs()[operatorGraphOutputEdge.GraphOutputIndex];
                 if (arg->Exists())
                 {
-                    nameToNodeAndIndexMap[arg->Name()] = NodeAndIndex{ nodeIndex, outputIndex };
+                    nameToNodeAndIndexMap[arg->Name()] = NodeAndIndex { 
+                        operatorGraphNodeIndexToMainGraphNodeIndexMap[operatorGraphOutputEdge.FromNodeIndex], 
+                        operatorGraphOutputEdge.FromNodeOutputIndex };
                 }
             }
 
-            DML_OPERATOR_DESC dmlDesc = SchemaHelpers::ConvertOperatorDesc(opDesc, &allocator);
-
-            ComPtr<IDMLOperator> op;
-            ORT_THROW_IF_FAILED(device->CreateOperator(&dmlDesc, IID_PPV_ARGS(&op)));
-            allocator.Reset();
-
-            NodeInfo nodeInfo = {};
-            nodeInfo.op = std::move(op);
-            graphNodes.push_back(std::move(nodeInfo));
+            if (isNodeAsOpDesc)
+            {
+                for (auto& opDesc : (*graphNodeCreateInfo.nodesAsOperatorDesc))
+                {
+                    DML_OPERATOR_DESC dmlDesc = SchemaHelpers::ConvertOperatorDesc(*opDesc, &allocator);
+                    ComPtr<IDMLOperator> op;
+                    ORT_THROW_IF_FAILED(device->CreateOperator(&dmlDesc, IID_PPV_ARGS(&op)));
+                    allocator.Reset();
+
+                    NodeInfo nodeInfo = {};
+                    nodeInfo.op = std::move(op);
+                    graphNodes.push_back(std::move(nodeInfo));
+                }
+            }
         }
 
-        assert(graphNodes.size() == orderedNodeIndices.size());
```

Comment:
>assert(graphNodes.size() == orderedNodeIndices.size()); [](http://example.com/codeflow?start=8&length=55)

What changed to make this invalid?

Reply:
The kernel for a single onnx node (or operator) may, now, contain more than 1 DML nodes, if that kernel is implemented as a graph.